src/Controller/AuthenticationStatusController.php

```diff
@@ -46,41 +46,53 @@ public function __construct(
     #[Route(path: '/authentication/status', name: 'app_identity_authentication_status', methods: ['GET'])]
     public function __invoke(): JsonResponse
     {
-        $nameId = $this->authenticationService->getNameId();
-        $sari = $this->stateHandler->getRequestId();
-        $logger = WithContextLogger::from($this->logger, ['nameId' => $nameId, 'sari' => $sari]);
-        $logger->info('Request for authentication status');
-
-        if (!$this->authenticationService->authenticationRequired()) {
-            $logger->error('There is no pending authentication request from SP');
-            return $this->refreshAuthenticationPage();
-        }
-
-        $isAuthenticated = $this->tiqrService->isAuthenticated();
-
-        if ($isAuthenticated) {
-            $logger->info('Send json response is authenticated');
-
-            return $this->refreshAuthenticationPage();
-        }
-
-        if ($this->authenticationChallengeIsExpired()) {
-            return $this->timeoutNeedsManualRetry();
+        try {
+            $nameId = $this->authenticationService->getNameId();
+            $sari = $this->stateHandler->getRequestId();
+            $logger = WithContextLogger::from($this->logger, ['nameId' => $nameId, 'sari' => $sari]);
+            $logger->info('Request for authentication status');
+
+            if (!$this->authenticationService->authenticationRequired()) {
+                $logger->error('There is no pending authentication request from SP');
+                return $this->refreshAuthenticationPage();
+            }
+
+            $isAuthenticated = $this->tiqrService->isAuthenticated();
+
+            if ($isAuthenticated) {
+                $logger->info('Send json response is authenticated');
+
+                return $this->refreshAuthenticationPage();
+            }
+
+            if ($this->authenticationChallengeIsExpired()) {
+                return $this->timeoutNeedsManualRetry();
+            }
+
+            $logger->info('Send json response is not authenticated');
+
+            return $this->scheduleNextPollOnAuthenticationPage();
+        } catch (Exception $e) {
+            $this->logger->error(
+                sprintf(
+                    'An unexpected authentication error occurred. Responding with "invalid-request", '.
+                    'original exception message: "%s"',
+                    $e->getMessage()
+                )
+            );
+            return $this->generateAuthenticationStatusResponse('invalid-request');
         }
-
-        $logger->info('Send json response is not authenticated');
-
-        return $this->scheduleNextPollOnAuthenticationPage();
     }
 
     /**
      * Generate a status response for authentication.html.
      *
-     * The javascript in the authentication page expects one of three statuses:
+     * The javascript in the authentication page expects one of four statuses:
```

Comment:
In my opinion these status codes are too tightly coupled with the javascript authentication page. The way I see it is that an API should be able to return any response (code) it likes and the client (in this case JS app) should be able to handle those situations.

Have you considered to add a default case in the statusReceivedHandler function in the JS app, instead of specifically handling the "invalid-request" response?

Also I think it'd be better to work with the status codes that belong to the type of error that happened. As far as I understand all of these errors now result in a 200 status code, while for example a 400 for the "invalid-request" would be better.